src/coreclr/gc/gc.cpp

```diff
@@ -36351,7 +36514,12 @@ BOOL gc_heap::decide_on_compacting (int condemned_gen_number,
         should_compact = TRUE;
     }
 
-#ifndef USE_REGIONS
+#ifdef USE_REGIONS
+    if (!should_compact)
+    {
+        should_compact = !!decide_on_compaction_space();
```

Comment:
it appears `decide_on_compaction_space` returns a `bool` so is `!!` required?

Reply:
I just put it there as a habit whenever the type doesn't match just that I know it's correct and I don't need to care about the details of how the conversion is done.